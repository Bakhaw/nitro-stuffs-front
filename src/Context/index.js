import React, { createContext, useEffect, useState } from 'react';

export const StateContext = createContext();

/** App State Provider used now only to save the current game (ex: csgo) */
export const StateProvider = ({ children }) => {
  const defaultGame = localStorage.getItem('currentGame');
  const [currentGame, setCurrentGame] = useState(defaultGame || 'csgo');

  const toggleGame = game => {
    localStorage.setItem('currentGame', game);
    setCurrentGame(game);
  };

  useEffect(() => {}, [currentGame]);

  return (
    <StateContext.Provider value={{ currentGame, toggleGame }}>
      {children}
    </StateContext.Provider>
  );
};
