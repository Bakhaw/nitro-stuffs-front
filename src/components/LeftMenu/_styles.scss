.LeftMenu_button {
  cursor: pointer;
  background: none;
  border: none;
  margin: 10px;

  &__img {
    fill: $White;
    width: 40px;
    height: 40px;
  }
}

.LeftMenu {
  background: $Blue;
  width: $LeftMenu__Width;
  height: $LeftMenu__Height;

  &__links {
    display: flex;
    flex-direction: column;
    align-items: center;
    padding-top: 35px;

    &__link {
      padding-bottom: 45px;
    }

    &__img {
      width: 60px;
      height: 60px;
      transition: 0.2s;

      &:hover {
        transform: scale(1.1);
        transition: 0.2s;
      }
    }
  }
}
