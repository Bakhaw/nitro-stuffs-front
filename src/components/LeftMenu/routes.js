const routes = [
  {
    img: 'csgo',
    to: '/game/csgo'
  },
  {
    img: 'valorant',
    to: '/game/valorant'
  }
];

export default routes;
