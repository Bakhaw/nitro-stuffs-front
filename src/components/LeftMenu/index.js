import React, { useContext, useState } from 'react';
import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
import { Link } from 'react-router-dom';

import Image from '../Image';
import { StateContext } from '../../Context';
import routes from './routes';

function LeftMenu() {
  const { toggleGame } = useContext(StateContext);
  const [open, setOpen] = useState(false);

  const toggleDrawer = (open) => () => {
    setOpen(open);
  };

  const onGameClick = (game) => () => {
    setOpen(false);
    toggleGame(game);
  };

  return (
    <nav>
      <button className='LeftMenu_button' onClick={toggleDrawer(true)}>
        <Image width='55' height='55' iconType='controller' />
      </button>
      <SwipeableDrawer
        disableDiscovery
        anchor='left'
        open={open}
        onClose={toggleDrawer(false)}
        onOpen={toggleDrawer(true)}
      >
        <div className='LeftMenu'>
          <ul className='LeftMenu__links'>
            {routes.map(({ gameKey, to }, index) => (
              <li className='LeftMenu__links__link' key={index}>
                <Link onClick={onGameClick(gameKey)} to={to}>
                  <img
                    alt={gameKey}
                    className='LeftMenu__links__img'
                    src={require(`../../static/images/${gameKey}/game-icon/${gameKey}.png`)}
                  />
                </Link>
              </li>
            ))}
          </ul>
        </div>
      </SwipeableDrawer>
    </nav>
  );
}

export default LeftMenu;
