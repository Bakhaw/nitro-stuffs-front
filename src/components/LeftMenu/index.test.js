import React from 'react';
import { shallow } from 'enzyme';

import { StateProvider } from '../../Context';

import LeftMenu from '.';

test('should render properly', () => {
  const container = shallow(
    <StateProvider>
      <LeftMenu />
    </StateProvider>
  );

  expect(container.debug()).toMatchSnapshot();
});
