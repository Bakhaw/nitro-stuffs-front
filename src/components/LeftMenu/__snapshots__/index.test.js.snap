// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render properly 1`] = `
"<Fragment>
  <button className=\\"LeftMenu_button\\" onClick={[Function]}>
    <img className=\\"LeftMenu_button__img\\" alt=\\"ControllerIcon\\" src=\\"controller.svg\\" />
  </button>
  <ForwardRef(SwipeableDrawer) disableDiscovery={true} anchor=\\"left\\" open={false} onClose={[Function]} onOpen={[Function]}>
    <div className=\\"LeftMenu\\">
      <ul className=\\"LeftMenu__links\\">
        <li className=\\"LeftMenu__links__link\\">
          <Link onClick={[Function]} to=\\"/game/csgo\\">
            <img alt=\\"csgo\\" className=\\"LeftMenu__links__img\\" src=\\"csgo.png\\" />
          </Link>
        </li>
        <li className=\\"LeftMenu__links__link\\">
          <Link onClick={[Function]} to=\\"/game/valorant\\">
            <img alt=\\"valorant\\" className=\\"LeftMenu__links__img\\" src=\\"valorant.png\\" />
          </Link>
        </li>
      </ul>
    </div>
  </ForwardRef(SwipeableDrawer)>
</Fragment>"
`;
